src/vmm/src/device_manager/legacy.rs

```diff
@@ -113,8 +105,27 @@ impl PortIODeviceManager {
 
     /// Register supported legacy devices.
     pub fn register_devices(&mut self, vm_fd: &VmFd) -> Result<()> {
-        let serial_2_4 = create_serial(self.com_evt_2_4.try_clone()?)?;
-        let serial_1_3 = create_serial(self.com_evt_1_3.try_clone()?)?;
+        let serial_2_4 = Arc::new(Mutex::new(BusDevice::Serial(SerialDevice {
```

Comment:
In this case where `create_serial` is a relatively small function, I prefer duplicating ~6 lines to avoid the extra indirection and complexity of having this function.

Reply:
There are well known issues having duplicate code. Many times its hard to know when its ok to have duplicate code and when its not so the standard practise is to follow the rule of 3 https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming). Since we are having duplicate code 3 times here as per the rule we should avoid it.

Reply:
This code is only used twice in production code, it is only in a test it is used a 3rd time. Given how small the code is, I would still argue it is easier to read being inline.